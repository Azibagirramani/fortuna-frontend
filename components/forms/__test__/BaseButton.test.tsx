import React from 'react';
import { render } from '@testing-library/react';

import BaseButton from '../BaseButton';

describe('BaseButton', () => {
    it('should render correctly', () => {
        const { container } = render(<BaseButton />);
        expect(container).toMatchSnapshot();
    });

    it('Should render with label in it', () => {
        const label: string = "Test Label";
        const { getAllByText } = render(<BaseButton>{label}</BaseButton>);
        expect(getAllByText(label)).toBeTruthy();
    });

    it("should display spinner when loading", () => {
        const isLoading: boolean = true;
        const { container } = render(<BaseButton loading={isLoading} />);
        expect(container.querySelector("#spinner")).toBeTruthy();
    });

    it("should receive all props", () => {

        const label: string = "Test Label";
        const disabled: boolean = true;
        const type: string = "submit";
        const onClick: () => void = () => { };
        const exClass: string = "test-class";
        const buttonId = "test-id";
        const { container } = render(<BaseButton id={buttonId} disabled={disabled} type={type} onClick={onClick} exClass={exClass}>{label}</BaseButton>);

        expect(container.querySelector(`#${buttonId}`)).toBeTruthy();
        expect(container.querySelector(`#${buttonId}`)).toHaveAttribute("id", buttonId);
        expect(container.querySelector(`#${buttonId}`)).toHaveAttribute("type", "submit");
        expect(container.querySelector(`#${buttonId}`)).toHaveAttribute("class", `py-5 xl:w-fit lg:w-full md:w-fit px-5 mr-2 text-sm font-medium text-white bg-secondary rounded shadow drop-shadow-sm test-class`);
        expect(container.querySelector(`#${buttonId}`)).toHaveTextContent(label);
    })


}
);