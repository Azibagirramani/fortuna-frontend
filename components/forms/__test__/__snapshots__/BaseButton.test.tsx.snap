// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton should render correctly 1`] = `
<div>
  <button
    class="py-5 xl:w-fit lg:w-full md:w-fit px-5 mr-2 text-sm font-medium text-white bg-secondary rounded shadow drop-shadow-sm undefined"
  />
</div>
`;
